File: problem/Min_Deletions/minimumDeletions.js
/**
 * @param {string} s
 * @return {number}
 */
const minimumDeletions = s => {
    let res = 0;
    let b = 0;

    for (const c of s) {
        if (c === 'b')
            b++;
        else if (b) {
            res++;
            b--;
        }
    }

    return res;
};

File: problem/Min_Deletions/minimumDeletions.c
int minimumDeletions(char *s)
{
    int res = 0, b = 0;

    for (int i = 0; s[i] != '\0'; i++)
    {
        if (~s[i] & 1)
        {
            b++;
        }
        else if (b)
        {
            res++;
            b--;
        }
    }

    return res;
}